xcm/xcm-builder/src/location_conversion.rs

```diff
@@ -473,6 +514,90 @@ mod tests {
 		assert_eq!(inverted, Err(()));
 	}
 
+	#[test]
+	fn global_consensus_converts_for_works() {
+		parameter_types! {
+			pub UniversalLocation: InteriorMultiLocation = X2(GlobalConsensus(ByGenesis([9; 32])), Parachain(1234));
+		}
+
+		let test_data = vec![
+			(MultiLocation::parent(), false),
+			(MultiLocation::new(0, Here), false),
+			(MultiLocation::new(0, X1(GlobalConsensus(ByGenesis([9; 32])))), false),
+			(MultiLocation::new(1, X1(GlobalConsensus(ByGenesis([9; 32])))), false),
+			(MultiLocation::new(2, X1(GlobalConsensus(ByGenesis([9; 32])))), false),
+			(MultiLocation::new(0, X1(GlobalConsensus(ByGenesis([0; 32])))), false),
+			(MultiLocation::new(1, X1(GlobalConsensus(ByGenesis([0; 32])))), false),
+			(MultiLocation::new(2, X1(GlobalConsensus(ByGenesis([0; 32])))), true),
+			(
+				MultiLocation::new(0, X2(GlobalConsensus(ByGenesis([0; 32])), Parachain(1000))),
+				false,
+			),
+			(
+				MultiLocation::new(1, X2(GlobalConsensus(ByGenesis([0; 32])), Parachain(1000))),
+				false,
+			),
+			(
+				MultiLocation::new(2, X2(GlobalConsensus(ByGenesis([0; 32])), Parachain(1000))),
+				false,
+			),
+		];
+
+		for (location, expected_result) in test_data {
+			let result =
+				GlobalConsensusConvertsFor::<UniversalLocation, [u8; 32]>::convert_location(
+					&location,
+				);
+			match result {
+				Some(account) => {
+					assert_eq!(
+						true, expected_result,
+						"expected_result: {}, but conversion passed: {:?}, location: {:?}",
+						expected_result, account, location
+					);
+					match &location {
+						MultiLocation { interior: X1(GlobalConsensus(network)), .. } =>
+							assert_eq!(
+								account,
+								GlobalConsensusConvertsFor::<UniversalLocation, [u8; 32]>::from_params(network),
+								"expected_result: {}, but conversion passed: {:?}, location: {:?}", expected_result, account, location
+							),
+						_ => assert_eq!(
+							true,
+							expected_result,
+							"expected_result: {}, conversion passed: {:?}, but MultiLocation does not match expected pattern, location: {:?}", expected_result, account, location
```

Comment:
This assert is already done in the start of the `Some(...)` branch. Do you need it? You could have an `if let` instead of a `match` if you don't.

Maybe it even makes sense to panic if the pattern is not `X1(GlobalConsensus(network))`, unless I'm missing something.

Reply:
ok, you're right, changed to panic